aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/CredentialProvidersDecorator.kt

```diff
@@ -24,48 +21,41 @@ import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType.Companion.pre
 import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
 import software.amazon.smithy.rust.codegen.core.smithy.customize.AdHocCustomization
 import software.amazon.smithy.rust.codegen.core.smithy.customize.adhocCustomization
-import software.amazon.smithy.rust.codegen.core.util.letIf
 
-class CredentialsProviderDecorator : ClientCodegenDecorator {
-    override val name: String = "CredentialsProvider"
-    override val order: Byte = 0
+class CredentialsProviderDecorator : ConditionalDecorator(
+    predicate = { codegenContext, _ -> codegenContext?.usesSigAuth() ?: false },
+    delegateTo =
+        object : ClientCodegenDecorator {
```

Comment:
I wonder if we can make this slightly cleaner like:

```
private class Unfiltered(....)

class CredentialsProviderDecorator: ConditionalDecorator(Unfiltered.Filter, Unfiltered())
```

might be a bit cleaner / less nesting? anyway, up to you